planner/core/find_best_task.go

```diff
@@ -835,14 +836,29 @@ func (is *PhysicalIndexScan) initSchema(idx *model.IndexInfo, idxExprCols []*exp
 			}
 		}
 	}
+
+	if is.Table.IsCommonHandle {
```

Comment:
Do we need to check `!is.Table.IsCommonHandle` above to avoid `setHandle`?

Reply:
CommonHandle table will not set `setHandle` to true because `PKIsHandle` is false and there is not `ExtraHandleID` column.
We can refine the code in the future to make the logic more clear.